File: golang/790.domino-and-tromino-tiling.go
const MOD = 1e9 + 7

func numTilings(n int) int {
	f, f1, f2, f3 := 1, 1, 1, 0

	for range n - 1 {
		f = (2*f1 + f3) % MOD
		f3 = f2
		f2 = f1
		f1 = f
	}

	return f
}

File: python/790.domino-and-tromino-tiling.py
class Solution:
    def numTilings(self, n: int) -> int:
        MOD = 10**9 + 7
        f = 1
        f_1 = 1
        f_2 = 1
        f_3 = 0

        for _ in range(n - 1):
            f = (2 * f_1 + f_3) % MOD
            f_3 = f_2
            f_2 = f_1
            f_1 = f

        return f

